airbyte-server/src/main/java/io/airbyte/server/handlers/SourceDefinitionsHandler.java

```diff
@@ -24,52 +24,102 @@
 
 package io.airbyte.server.handlers;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.util.ClassUtil;
+import com.google.common.base.Preconditions;
 import io.airbyte.api.model.SourceDefinitionCreate;
 import io.airbyte.api.model.SourceDefinitionIdRequestBody;
 import io.airbyte.api.model.SourceDefinitionRead;
 import io.airbyte.api.model.SourceDefinitionReadList;
 import io.airbyte.api.model.SourceDefinitionUpdate;
+import io.airbyte.commons.yaml.Yamls;
 import io.airbyte.config.StandardSourceDefinition;
+import io.airbyte.config.init.SeedRepository;
 import io.airbyte.config.persistence.ConfigNotFoundException;
 import io.airbyte.config.persistence.ConfigRepository;
 import io.airbyte.scheduler.client.CachingSchedulerJobClient;
+import io.airbyte.server.errors.KnownException;
 import io.airbyte.server.validators.DockerImageValidator;
 import io.airbyte.validation.json.JsonValidationException;
 import java.io.IOException;
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.util.ArrayList;
+import java.util.Iterator;
 import java.util.List;
 import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class SourceDefinitionsHandler {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(SourceDefinitionsHandler.class);
+  private static final String DEFAULT_SOURCE_DEFINITION_ID_NAME = "sourceDefinitionId";
```

Comment:
Some of these stuff can be potentially pulled out into either a `JavaHttpClient` class, or a 'Github source of truth' model. Felt unnecessary given that most of our stuff I expect only two usages - SourceDefinition and DesintationDefinition in the short/medium term.

Reply:
I feel combining them now is slightly premature. IMO we don't get much yet since the functionality is nicely encapsulated in 2 - 3 functions within each file. I'd rather wait for when we have a 3rd use case to do it.

You have a better sense of the codebase + future use cases so I'll defer to you. If using Github as a data store is something we foresee and want to encourage then I feel creating a separate concept makes sense.

Ultimately I'm a 2/10 on this one so if you feel stronger I'm happy to do so!